spec/nexmo-jwt/jwt_builder_spec.rb

```diff
@@ -155,4 +184,17 @@
       }.to raise_error(KeyError, "key not found: :private_key")
     end
   end
+
+  context 'with conflicting parameters' do
+    it 'raises a custom exception if ttl and exp params both provided' do
+      expect {
+        builder = Nexmo::JWTBuilder.new(
+          application_id: '123456789',
+          private_key: './spec/nexmo-jwt/private_key.txt',
+          ttl: 500,
+          exp: 600
+        )
+      }.to raise_error(ArgumentError, "Expected either 'ttl' or 'exp' parameter not both")
```

Comment:
Should we recommend setting TTL over exp?

Reply:
For consistency with the spec, yes, the user should set TTL. Saying, "Expire in 15 minutes" is easier and more clear than *do some math* "Expire at 9999999999". It also avoid a misconception if the generator is used in a long running process, because TTL is always applied to IAT, versus Expiration which is always a concrete number.

Since the original Ruby SDK supported setting a hard-coded EXP, we can let that slide as an immediate deprecation as we update the docs and move users away.